packages/backend/src/controllers/authentication.ts

```diff
@@ -73,6 +76,150 @@ export const getOpenIdUserFromRequest = ({
     return undefined;
 };
 
+const createOpenIdUser = (
+    userInfo: UserinfoResponse | undefined,
+    profile: PassportProfile | undefined,
+    issuer: string,
+    issuerType: OpenIdIdentityIssuerType,
+): OpenIdUser => {
+    const email = userInfo?.email || profile?.emails?.[0]?.value;
+    const subject = userInfo?.sub || profile?.id;
+    if (!(email && subject)) {
+        throw new Error('Could not parse authentication token');
+    }
+
+    const displayName = userInfo?.name || profile?.displayName || '';
+    const [fallbackFirstName, fallbackLastName] = displayName.split(' ');
+    const firstName =
+        userInfo?.given_name || profile?.name?.givenName || fallbackFirstName;
+    const lastName =
+        userInfo?.family_name || profile?.name?.familyName || fallbackLastName;
+
+    const openIdUser: OpenIdUser = {
+        openId: {
+            issuer: issuer || '',
+            email,
+            subject,
+            firstName,
+            lastName,
+            issuerType: issuerType || '',
+        },
+    };
+    if (userInfo?.groups && Array.isArray(userInfo.groups)) {
+        openIdUser.openId.groups = userInfo.groups;
+    }
+
+    return openIdUser;
+};
+
+const setupClient = async () => {
+    const oktaIssuer = await Issuer.discover(
+        `https://${lightdashConfig.auth.okta.oktaDomain}/oauth2/default`,
+    );
+
+    const redirectUri = new URL(
+        `/api/v1${lightdashConfig.auth.okta.callbackPath}`,
+        lightdashConfig.siteUrl,
+    ).href;
+
+    const client = new oktaIssuer.Client({
+        client_id: lightdashConfig.auth.okta.oauth2ClientId ?? '',
```

Comment:
Did you test this when okta isn't setup? I'm expecting that the passport middleware is simply not mounted based on the code further down in `index.ts`

Reply:
Looking into this!

Reply:
oh, it won't even work because of the line in: https://github.com/lightdash/lightdash/pull/8508/files#r1445921312